ext/storage_sources/local_store/local_store.c

```diff
@@ -0,0 +1,1220 @@
+      strlen(name) + 2;
+    if ((p = malloc(len)) == NULL)
+        return (local_err(local, NULL, ENOMEM, "local_location_path"));
+    snprintf(p, len, "%s/%s%s%s", location->bucket, marker, location->cluster_prefix, name);
+    *pathp = p;
+    return (ret);
+}
+
+/*
+ * local_exist --
+ *     Return if the file exists.
+ */
+static int
+local_exist(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
+  WT_LOCATION_HANDLE *location_handle, const char *name, bool *existp)
+{
+    struct stat sb;
+    LOCAL_STORAGE *local;
+    int ret;
+    char *path;
+
+    local = (LOCAL_STORAGE *)storage_source;
+    path = NULL;
+
+    local->op_count++;
+    if ((ret = local_location_path(local, location_handle, name, NULL, &path)) != 0)
+        goto err;
+
+    ret = stat(path, &sb);
+    if (ret == 0)
+        *existp = true;
+    else if (errno == ENOENT) {
+        ret = 0;
+        *existp = false;
+    } else
+        ret = local_err(local, session, errno, "%s: ss_exist stat", path);
+
+err:
+    free(path);
+    return (ret);
+}
+
+/*
+ * local_flush --
+ *     Return when the files have been flushed.
+ */
+static int
+local_flush(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
+  WT_LOCATION_HANDLE *location_handle, const char *name, const char *config)
+{
+    LOCAL_STORAGE *local;
+    LOCAL_FLUSH_ITEM *flush, *safe_flush;
+    int ret, t_ret;
+    char *match;
+
+    (void)config; /* Unused */
+
+    /*
+     * This implementation does not do anything meaningful on flush. However, we do track which
+     * objects have not yet been flushed and note which ones need to be flushed now.
+     */
+    ret = 0;
+    local = (LOCAL_STORAGE *)storage_source;
+    match = NULL;
+
+    if (location_handle == NULL && name != NULL)
+        return local_err(local, session, EINVAL, "flush: cannot specify name without location");
+
+    local->op_count++;
+    if (location_handle != NULL) {
+        if ((ret = local_location_path(
+               local, location_handle, name == NULL ? "" : name, NULL, &match)) != 0)
+            goto err;
+    }
+    VERBOSE(local, "Flush: match=%s\n", SHOW_STRING(match));
+
+    /*
+     * Note: we retain the lock on the data structure while flushing all entries. This is fine for
+     * our local file implementation, when we don't have to do anything to flush, but for a cloud
+     * implementation, we'll want some way to not hold the lock while transferring data.
+     */
+    if ((ret = pthread_rwlock_wrlock(&local->flush_lock)) != 0) {
+        (void)local_err(local, session, ret, "flush: pthread_rwlock_wrlock");
+        goto err;
+    }
+
+    TAILQ_FOREACH_SAFE(flush, &local->flushq, q, safe_flush)
+    {
+        if (match != NULL) {
+            /* Make sure the location matches. */
+            if (name != NULL) {
+                /* Exact name match required. */
+                if (strcmp(flush->src_path, match) != 0)
```

Comment:
This is confusing and I'm not sure if it is really a bug. We're testing `name != NULL` and then don't use `name` at all here. It is basically a boolean and that is unexpected. `match` is set to a string that ends in `name`. So won't the string be "somestringNAME" or "somesting" depending on `name`? Can we do `strcmp` for both?

Reply:
I think the code is correct.  I added a comment that I think will clarify what's going on.